src/engraving/rw/compat/compatutils.cpp

```diff
@@ -341,6 +342,78 @@ void CompatUtils::reconstructTypeOfCustomDynamics(MasterScore* score)
     }
 }
 
+void CompatUtils::splitArticulations(MasterScore* masterScore)
+{
+    std::set<Articulation*> toRemove;
+    for (Score* score : masterScore->scoreList()) {
+        for (Measure* meas = score->firstMeasure(); meas; meas = meas->nextMeasure()) {
+            for (Segment& seg : meas->segments()) {
+                if (!seg.isChordRestType()) {
+                    continue;
+                }
+                for (EngravingItem* item : seg.elist()) {
+                    if (!item || !item->isChord()) {
+                        continue;
+                    }
+                    Chord* chord = toChord(item);
+                    for (Articulation* a : chord->articulations()) {
+                        std::set<SymId> ids = mu::engraving::splitArticulations({ a->symId() });
+                        if (ids.size() <= 1) {
+                            continue;
+                        }
+                        toRemove.insert(a);
+                    }
+                }
+            }
+        }
+        // separate into individual articulations
+        for (Articulation* combinedArtic : toRemove) {
+            auto components = mu::engraving::splitArticulations({ combinedArtic->symId() });
+            Chord* parentChord = toChord(combinedArtic->parentItem());
+            for (SymId id : components) {
+                Articulation* newArtic = Factory::createArticulation(score->dummy()->chord());
+                newArtic->setSymId(id);
+                if (parentChord->hasArticulation(newArtic)) {
+                    delete newArtic;
+                    continue;
+                }
+                newArtic->setParent(parentChord);
+                newArtic->setTrack(combinedArtic->track());
+                newArtic->setPos(combinedArtic->pos());
+                newArtic->setDirection(combinedArtic->direction());
+                newArtic->setAnchor(combinedArtic->anchor());
+                if (!parentChord->isLinked() && score == masterScore) {
+                    // need to create a new Links object for the new artic
+                    LinkedObjects* links = new LinkedObjects(masterScore);
+                    links->push_back(newArtic);
+                    newArtic->setLinks(links);
+                } else {
+                    // link to the existing Links object
+                    Chord* mainChord = toChord(parentChord->links()->mainElement());
+                    // find the corresponding articulation
+                    Articulation* mainArtic = nullptr;
+                    for (Articulation* artic : mainChord->articulations()) {
+                        if (artic->symId() == id) {
```

Comment:
This assumes that we will always see the main element first (and its linked elements after that), but I'm afraid this is not guaranteed. 

What do you think about the following strategy:
1. Use the master score to collect all articulations that need to be split _and are `!isLinked()`_ 
2. For each of the collected main articulations:
    1. For each of the component SymIds for that articulation:
        1. Create the new main articulation for that component
        2. For all linked elements of the original main articulation, create a new articulation for the component, and link it to the new main articulation for that componenti.e.
            ```c++
                    LinkedObjects* links = originalMainArticulation->links();
                    if (!links || links->empty()) {
                        continue;
                    }
                    for (EngravingObject* originalLinkedItem : *links) {
                        if (originalLinkedItem != originalMainArticulation) {
                            // Create a new articulation for the component and link it to the new main articulation for the component
                        }
                    }
            ```
    2. After all new component articulations have been created, the original ones can be deleted (so, delete the main one _and_ its linked ones)

This way, you don't need to use scoresList(), and you don't need to look up the new articulation in the parentChord's articulations.